src/chain/elders_info.rs

```diff
@@ -80,25 +80,32 @@ impl EldersInfo {
 
     /// Creates a new `EldersInfo` by merging this and the other one.
     pub fn merge(&self, other: &Self) -> Result<Self, RoutingError> {
-        let members = self.members.iter().chain(&other.members).cloned().collect();
+        let members = self
+            .members
+            .iter()
+            .chain(&other.members)
+            .map(|(pub_id, p2p_node)| (*pub_id, p2p_node.clone()))
+            .collect();
         Self::new(members, self.prefix.popped(), vec![self, other])
     }
 
     // WIP: remove me (or make less heavy)
     pub fn members(&self) -> BTreeSet<PublicId> {
-        self.members
-            .iter()
-            .map(P2pNode::public_id)
-            .copied()
-            .collect()
+        self.members.keys().copied().collect()
     }
 
-    pub fn p2p_members(&self) -> &BTreeSet<P2pNode> {
+    // WIP: remove me?
+    // WIP: rename to member_map?
+    pub fn p2p_members(&self) -> &BTreeMap<PublicId, P2pNode> {
         &self.members
     }
 
+    pub fn member_nodes(&self) -> impl Iterator<Item = &P2pNode> {
+        self.members.values()
+    }
+
     pub fn member_names(&self) -> BTreeSet<XorName> {
```

Comment:
Probably same here `member_names_as_set`.
Also should it better be an iterator that we return? (it depends how often it is used and how).

Reply:
Agree, whenever we copy/clone to a new container it makes sense to indicate it in the name